packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -201,6 +204,33 @@ export class SelectedNetworkController extends BaseController<
         });
       },
     );
+
+    onPreferencesStateChange(({ useRequestQueue }) => {
+      if (this.#useRequestQueuePreference !== useRequestQueue) {
+        if (!useRequestQueue) {
+          Object.keys(this.state.domains).forEach((domain) => {
+            this.#unsetNetworkClientIdForDomain(domain);
+          });
+        } else {
+          // eslint-disable-next-line
+          (this.#domainProxyMap as any).map.forEach(
```

Comment:
🤔 not sure the best way to make this not awkward and more agnostic to this custom class

Reply:
added [a pass through forEach on the WeakRefObjectMap class](https://github.com/MetaMask/metamask-extension/pull/23525/files#diff-4d13f1f0ba4d949f6dfe026c7c50a982dfd0e3b43d3bbd265f6acdca8f501302R57) for now